CMakeLists.txt

```diff
@@ -730,6 +730,10 @@ if(CONSTANT_TIME_VALIDATION)
   add_definitions(-DNDEBUG)
 endif()
 
+if(CMAKE_BUILD_TYPE_LOWER STREQUAL "debug")
```

Comment:
(See other comment.) 

There are certain common macros that are specific to debug/release builds that we could use to automatically detect a debug build.
```C++
#if defined(DEBUG) || defined(_DEBUG) || !defined(NDEBUG)
...
```